design/queries.go

```diff
@@ -0,0 +1,120 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var query = a.Type("Query", func() {
+	a.Description(`JSONAPI store for the data of a Query. See also http://jsonapi.org/format/#document-resource-object`)
+	a.Attribute("type", d.String, func() {
+		a.Enum("queries")
+	})
+	a.Attribute("id", d.UUID, "ID of query", func() {
+		a.Example("40bbdd3d-8b5d-4fd6-ac90-7236b669af04")
+	})
+	a.Attribute("attributes", queryAttributes)
+	a.Attribute("links", genericLinks)
+	a.Attribute("relationships", queryRelationships)
+	a.Required("type", "attributes")
+})
+
+var queryRelationships = a.Type("QueryRelations", func() {
```

Comment:
query relationships should contain space (e.g. https://github.com/fabric8-services/fabric8-wit/blob/master/design/labels.go#L46)